src/vivarium_public_health/metrics/mortality.py

```diff
@@ -0,0 +1,137 @@
+import pandas as pd
+import numpy as np
+
+from .utilities import get_age_bins
+
+
+class MortalityObserver:
+    """ An observer for total and cause specific deaths during simulation.
+    This component by default observes the total number of deaths out of
+    total population and total person time that each simulant spent
+    during the simulation until it exits.
+    These data are categorized by age groups and causes and aggregated over
+    total population as well as the population who were born during
+    the simulation.
```

Comment:
I find this doc string kind of confusing. From the code, it seems like deaths and person time are always grouped by age (i.e., not aggregated over total population) but regardless of whether or not you have the by_year flag turned on, you will get deaths and person time aggregated over the whole course of the sim. If you have the by_year flag turned on, you will additionally get deaths and person time by year. Reading this doc string, I am confused about what results I'm going to get by what group. 

also: why do we only group by age/year and not sex? is that not of interest?